src/matchmaking-threads.ts

```diff
@@ -0,0 +1,144 @@
+import { MatchmakingThread } from '@/models/matchmakingThreads';
+import { User } from '@/models/users';
+import { sendEventLogMessage } from '@/util';
+import { getDB } from './db';
+import { ChannelType, EmbedBuilder } from 'discord.js';
+import type { Client, Message } from 'discord.js';
+
+export async function handleMatchmakingThreadMessage(message: Message): Promise<void> {
+	const matchmakingChannelId = getDB().get('channels.matchmaking');
+	if (!matchmakingChannelId) {
+		console.log('Missing matchmaking channel!');
+		return;
+	}
+
+	if (message.channel.type !== ChannelType.PublicThread || message.channel.parentId !== matchmakingChannelId) {
+		return;
+	}
+
+	await MatchmakingThread.upsert({ thread_id: message.channelId, last_message_sent: message.createdAt });
+}
+
+export async function checkMatchmakingThreads(client: Client): Promise<void> {
```

Comment:
Why pass the client and not just import it?

Reply:
Where would I import it from?

Reply:
From `bot.ts`. It's not exported at the moment, but it can be if it's useful

Either way is fine, to be clear. I was asking if there was any reason to not just export it

Reply:
No specific reason then, I just did this because it seemed simplest. I'll change it if that is useful.